code/game/objects/items/melee/batons/classic.dm

```diff
@@ -0,0 +1,133 @@
+		else
+			user.take_bodypart_damage(2*force)
+		return TRUE
+	if(iscyborg(target))
+		return FALSE
+	if(!isliving(target))
+		return FALSE
+	if (user.a_intent == INTENT_HARM)
+		return FALSE
+	else
+		if(last_hit < world.time)
+			if(target.check_shields(src, 0, "[user]'s [name]", MELEE_ATTACK))
+				playsound(target, 'sound/weapons/genhit.ogg', 50, 1)
+				return TRUE
+			if(ishuman(target))
+				var/mob/living/carbon/human/H = target
+				if(check_martial_counter(H, user))
+					return TRUE
+			playsound(get_turf(src), 'sound/effects/woodhit.ogg', 75, 1, -1)
+			target.Knockdown(softstun_ds, TRUE, FALSE, hardstun_ds, stam_dmg)
+			log_combat(user, target, "stunned", src)
+			src.add_fingerprint(user)
+			target.visible_message("<span class ='danger'>[user] has knocked down [target] with [src]!</span>", \
+				"<span class ='userdanger'>[user] has knocked down [target] with [src]!</span>")
+			if(!iscarbon(user))
+				target.LAssailant = null
+			else
+				target.LAssailant = user
+			last_hit = world.time + cooldown
+			user.adjustStaminaLossBuffered(getweight())//CIT CHANGE - makes swinging batons cost stamina
+	return TRUE
+
+/obj/item/melee/classic_baton/telescopic
+	name = "telescopic baton"
+	desc = "A compact yet robust personal defense weapon. Can be concealed when folded."
+	icon = 'icons/obj/items_and_weapons.dmi'
+	icon_state = "telebaton_0"
+	lefthand_file = 'icons/mob/inhands/weapons/melee_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/melee_righthand.dmi'
+	item_state = null
+	slot_flags = ITEM_SLOT_BELT
+	w_class = WEIGHT_CLASS_SMALL
+	item_flags = NONE
+	force = 0
+	total_mass = TOTAL_MASS_NORMAL_ITEM
+
+/obj/item/melee/classic_baton/telescopic/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/transforming, ATTACK_SELF_TRANSFORM, .proc/on_transform)
+
+/obj/item/melee/classic_baton/baton_stun(mob/living/target, mob/living/user)
+	if(!(SEND_SIGNAL(src, COMSIG_IS_TRANSFORMED) & COMPONENT_IS_TRANSFORMED))
+		return FALSE
+	return ..()
+
+/obj/item/melee/classic_baton/telescopic/suicide_act(mob/user)
+	var/mob/living/carbon/human/H = user
+	var/obj/item/organ/brain/B = H.getorgan(/obj/item/organ/brain)
+
+	user.visible_message("<span class='suicide'>[user] stuffs [src] up [user.p_their()] nose and presses the 'extend' button! It looks like [user.p_theyre()] trying to clear [user.p_their()] mind.</span>")
+	SEND_SIGNAL(src, COMSIG_TRANSFORM_ACTIVATE, user)
+	add_fingerprint(user)
+	sleep(3)
+	if (H && !QDELETED(H))
+		if (B && !QDELETED(B))
+			H.internal_organs -= B
+			qdel(B)
+		H.spawn_gibs()
+		return (BRUTELOSS)
+
+/obj/item/melee/classic_baton/telescopic/proc/on_transform(obj/item/parent, new_active, mob/user, forced, datum/signal_source, checks_passed)
+	if(user)
+		if(new_active)
+			to_chat(user, "<span class ='warning'>You extend the baton.</span>")
+		else
+			to_chat(user, "<span class ='notice'>You collapse the baton.</span>")
+	if(new_active)
```

Comment:
i'm not sure what i should do here

maybe i can make a subtype of /transforming component that auto sets these i guess if we uh, really want to reduce copy paste

ugh, i fucking hate designing sometimes.